### PR TITLE
Remove all long_vector_type uses

### PR DESCRIPTION
**Pull request by pinkwah**

### CODE REVIEW

**Review by oyvindeide — APPROVED** [review]
LGTM, just a style fix

File: libres/lib/res_util/block_fs.cpp [hunk 1]
```diff
@@ -864,15 +864,15 @@ static void block_fs_preload(block_fs_type *block_fs) { return; }
...
 
 static void block_fs_fix_nodes(block_fs_type *block_fs,
-                               long_vector_type *offset_list) {
+                               const std::vector<long>& offset_list) {
     if (block_fs->data_owner) {
         fsync(block_fs->data_fd);
         {
             char *key = NULL;
-            for (int inode = 0; inode < long_vector_size(offset_list);
+            for (int inode = 0; inode < offset_list.size();
```
Comment (eivindjahren) [inline libres/lib/res_util/block_fs.cpp:872]: I believe the c tests only call this function with offset_list.size() == 0 and so the brunt of this function only has integration tests.

File: libres/lib/res_util/block_fs.cpp [hunk 2]
```diff
@@ -964,7 +964,7 @@ static void block_fs_build_index(block_fs_type *block_fs,
                             "** Warning found node:%s at offset:%ld which was "
                             "incomplete - discarded.\n",
                             filename, file_node->node_offset);
-                    long_vector_append(error_offset, file_node->node_offset);
+                    error_offset.push_back(file_node->node_offset);
```
Comment (eivindjahren) [inline libres/lib/res_util/block_fs.cpp:967]: There is no coverage of this function in the c tests.

**Review by eivindjahren — COMMENTED** [review]
Just some comments about what part of the diff is covered by the c test suite, but perhaps not truly relevant for this change.

**Comment by eivindjahren** [comment]
I think just resolving the conflicts should make this PR mergable.

File: libres/lib/res_util/block_fs.cpp [hunk 3]
```diff
@@ -864,15 +864,14 @@ static void block_fs_preload(block_fs_type *block_fs) { return; }
...
 static void block_fs_fix_nodes(block_fs_type *block_fs,
-                               long_vector_type *offset_list) {
+                               const std::vector<long> &offset_list) {
     if (block_fs->data_owner) {
         fsync(block_fs->data_fd);
         {
             char *key = NULL;
-            for (int inode = 0; inode < long_vector_size(offset_list);
-                 inode++) {
+            for (int inode = 0; inode < offset_list.size(); inode++) {
```
Comment (oyvindeide) [inline libres/lib/res_util/block_fs.cpp:872]: Consider: `for (auto &node_offset : offset_list)`?

**Comment by eivindjahren** [comment]
Seems just a rebase away from being mergeable...

File: libres/lib/res_util/block_fs.cpp (hunk 3 above)
Comment (pinkwah) [reply libres/lib/res_util/block_fs.cpp]: Changed.

**Pull request merged**